codejar.ts

```diff
@@ -13,6 +13,8 @@ type Position = {
   dir?: "->" | "<-"
 }
 
+export type CodeJar = ReturnType<typeof CodeJar>;
```

Comment:
Awesome! But what if we instead of deleting `new` from docs will define `export type CodeJar` which is equals to something like `new ReturnType<typeof CodeJar>`?
I don't want to really change public API.

Reply:
Hmm... I'll see if there is a way to define the TS type for `CodeJar` in a way which will allow the "new" keyword..

Reply:
OK, please take a look at the latest commit, in which I use an approach that allows the "new CodeJar()" API in TypeScript.